src/plugins/data_view_field_editor/public/components/preview/field_preview_context.tsx

```diff
@@ -165,13 +166,21 @@ export const FieldPreviewProvider: FunctionComponent = ({ children }) => {
       if (format?.id) {
         const formatter = fieldFormats.getInstance(format.id, format.params);
         if (formatter) {
-          return formatter.convertObject?.html(value) ?? JSON.stringify(value);
+          return formatter.getConverterFor('html')(value) ?? JSON.stringify(value);
+        }
+      }
+
+      if (type) {
```

Comment:
Adding this case in case there is no `format` override selected. In this case we need to use default field format for current type 

fixes https://github.com/elastic/kibana/issues/118494